### PR TITLE
Initial pass of versioning guidelines

### PR DESCRIPTION
**Pull request by mikekistler**
This PR adds a new document focused specifically on best practices for versioning of Azure services.

### CODE REVIEW

File: azure/VersioningGuidelines.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Azure Versioning Guidelines
+
+## History
+
+<details>
+  <summary>Expand change history</summary>
+
+| Date        | Notes                                                          |
+| ----------- | -------------------------------------------------------------- |
+| 2024-Nov-14 | Updated LRO guidelines                                         |
```
Comment (JeffreyRichter) [inline azure/VersioningGuidelines.md]:
```suggestion
| 2024-Nov-14 | Azure Service Versioning & Breaking Change Guidelines       |
```

File: azure/VersioningGuidelines.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+</details>
+
+## Guidelines
+
+This document provides a "Dos and Don'ts" list for complying with the Azure Versioning and Breaking Change Policy,
+as documented [internally](aka.ms/AzBreakingChangesPolicy) and [externally](https://learn.microsoft.com/azure/developer/intro/azure-service-sdk-tool-versioning).
+
+:white_check_mark: **DO** thoroughly test the API as documented before merging to a production branch of the specs repo.
```
Comment (JeffreyRichter) [inline azure/VersioningGuidelines.md]:
```suggestion
:white_check_mark: **DO** thoroughly ensure/test the API contract is entirely correct before merging it into a production branch of the specs repo.
```

File: azure/VersioningGuidelines.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+Testing helps avoid "BugFix" changes to the API definition. Testing should be done at the HTTP level as well as through generated SDKs.
+
+:white_check_mark: **DO** retire all prior preview API versions when a new GA or preview API version is released.
```
Comment (JeffreyRichter) [inline azure/VersioningGuidelines.md]: Should we say anything about the 90 days period?

File: azure/VersioningGuidelines.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+:white_check_mark: **DO** contact the Azure Breaking Change Review board to coordinate communications to customers
+when releasing an API version that will require retiring a prior version.
```
Comment (JeffreyRichter) [inline azure/VersioningGuidelines.md]:
```suggestion
when releasing an API version requiring the retirement of a prior version.
```

**Review by JeffreyRichter — APPROVED** [review]
I made a few suggestions but otherwise, I approve

File: azure/VersioningGuidelines.md (hunk 3 above)
Comment (mikekistler) [inline azure/VersioningGuidelines.md]:
```suggestion
:white_check_mark: **DO** retire all prior preview API versions 90 days after a new GA or preview API version is released.
```

**Pull request merged**